xlators/cluster/ec/src/ec-types.h

```diff
@@ -562,30 +562,20 @@ struct _ec_matrix_list {
 };
 
 struct _ec_heal {
-    struct list_head list;
     gf_lock_t lock;
     xlator_t *xl;
     ec_fop_data_t *fop;
     void *data;
-    ec_fop_data_t *lookup;
     loc_t loc;
-    struct iatt iatt;
-    char *symlink;
+    ia_type_t ia_type;
```

Comment:
Nice cleanup! Can you ensure alignment? I use 'pahole' to look at the structure - padding and cache line alignment.
(could be a different patch of course)

Reply:
Well, further analysis seems shows that 'ia_type' is never changed after being initialized with IA_IFREG (and so redundantly checked in ec_heal_data_block()). If so, removing one more 4-byte field from 'struct _ec_heal' makes it's layout much better.